core/src/test/java/org/testcontainers/junit/DockerComposeContainerWithBuildTest.java

```diff
@@ -0,0 +1,85 @@
+package org.testcontainers.junit;
+
+import com.github.dockerjava.api.model.Container;
+import org.junit.Test;
+import org.rnorth.ducttape.unreliables.Unreliables;
+import org.testcontainers.DockerClientFactory;
+import org.testcontainers.containers.DockerComposeContainer;
+
+import java.io.File;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.stream.Stream;
+
+import static org.rnorth.visibleassertions.VisibleAssertions.*;
+import static org.testcontainers.containers.DockerComposeContainer.RemoveImages.ALL;
+import static org.testcontainers.containers.DockerComposeContainer.RemoveImages.LOCAL;
+
+public class DockerComposeContainerWithBuildTest {
+
+    @Test
+    public void defaultDoesNotRemoveAnyImages() {
+        doTest(null, true, true);
+    }
+
+    @Test
+    public void localModeRemovesBuiltImages() {
+        doTest(LOCAL, false, true);
+    }
+
+
+    @Test
+    public void allModeRemovesAllImages() {
+        doTest(ALL, false, false);
+    }
+
+    private void doTest(final DockerComposeContainer.RemoveImages removeMode,
```

Comment:
@rnorth WDYT about using the parameterized runner instead?

Reply:
May as well - done!